### PR TITLE
[NWPS-1127] Tagging plugin

### PR DESCRIPTION
**Pull request by sankar-shunmuga**
Please review this.
I included tags for all page layouts
image

### CODE REVIEW

File: repository-data/application/src/main/resources/hcm-config/configuration/translations/types.yaml [hunk 1]
```diff
@@ -15,11 +15,39 @@ definitions:
       /en:
         jcr:primaryType: hipposys:resourcebundle
         jcr:name: '[Page] Landing page'
+        hee:tags: Keywords
+        hee:tagssupport: Keyword suggestions
```
Comment (smanivasag) [inline repository-data/application/src/main/resources/hcm-config/configuration/translations/types.yaml:19]: Could we remove `hee:tagssupport` translation for all page document types (as apparently, it doesn't override the UI label/caption for the field)?

I think we may need to override it via page document type editor template configuration (e.g. `/hippo:namespaces/hee/<page_document_type>/editor:templates/_default_/tagsuggest/@caption='Tag suggestions'`) or override `hippo-plugin-content-tagging-14.7.13.jar!/org/onehippo/forge/ecmtagging/editor/TagSuggestPlugin.properties` resource bundle.

But, we may not be able to override the `Refresh keywords` label via the document type editor template configuration and so we may need to do it by overriding `hippo-plugin-content-tagging-14.7.13.jar!/org/onehippo/forge/ecmtagging/editor/TagSuggestPlugin.properties` resource bundle.

**Review by smanivasag — COMMENTED** [review]
@ianp-mani @sankar-shunmuga - Thanks for the PR 👍 

- Could we generate/add the following `getTags()` method for other page document type beans (e.g. `uk.nhs.hee.web.beans.News`, etc) similar to `uk.nhs.hee.web.beans.BlogPost` for completeness?
```
  @HippoEssentialsGenerated(internalName = "hippostd:tags")
  public String[] getTags() {
      return getMultipleProperty("hippostd:tags");
  }
```
- I guess `Bulletin` and `Searchbank` are also page document types similar to `Casestudy`. In that case, could we add tagging for them as well?
- We may need to add modified development data paths to `repository-data/development/src/main/resources/hcm-actions.yaml` to reload them for tagging.


Note: Have realised that it is possible to move tags (`Keywords`) & tagsuggest (`Keyword suggestions`) fields around a document type via the document type editor and so we can leverage it to place them in a different place within a document type (instead of at the very end of the document type) if needed.

File: site/components/src/main/java/uk/nhs/hee/web/beans/YellowAlertBlock.java
[the remark this answers is not in the document]
Comment (ianp-mani) [reply site/components/src/main/java/uk/nhs/hee/web/beans/YellowAlertBlock.java:1]: This keeps popping up in code I check out .. where did it originate ? @smanivasag

[the remark this answers is not in the document]
Comment (smanivasag) [reply site/components/src/main/java/uk/nhs/hee/web/beans/YellowAlertBlock.java:1]: I think it originates from `repository-data/application/src/main/resources/hcm-config/namespaces/hee-cms-platform.cnd`:
```
File: repository-data/application/src/main/resources/hcm-config/namespaces/hee-cms-platform.cnd
35: [hee:YellowAlertBlock] > hippo:compound, hippostd:relaxed
36:   orderable
```
As part of `NWPS-964`, we have removed all the references of the old warning callout block `hee:YellowAlertBlock` except from `repository-data/application/src/main/resources/hcm-config/namespaces/hee-cms-platform.cnd` as we understand it's a backward incompatible change as described in https://community.bloomreach.com/t/how-to-delete-compound-type/600 and https://community.bloomreach.com/t/delete-document-type/1833/3. We can perhaps check if there is a way to configure essentials bean generator tool not to generate one for `hee:YellowAlertBlock`. Thanks!

File: site/components/src/main/java/uk/nhs/hee/web/beans/PublicationResource.java
[the remark this answers is not in the document]
Comment (smanivasag) [reply site/components/src/main/java/uk/nhs/hee/web/beans/PublicationResource.java:1]: Again from `repository-data/application/src/main/resources/hcm-config/namespaces/hee-cms-platform.cnd`. Fyi - We still use this node type on the publication landing page (`hee:publicationLandingPage`) to configure upload file limits (file types, size, etc) for the `Document Versions` & `Alternate language versions` fields. But, I assume `hee:publicationResource` will be removed as part of `NWPS-1357`. Thanks!

**Pull request closed without merging**